lib/src/Base/Graph/GraphImplementation.cxx

```diff
@@ -144,7 +144,11 @@ String GraphImplementation::__repr__() const
 /* Adds a drawable instance to the collection of drawables contained in GraphImplementation */
 void GraphImplementation::add(const Drawable & aDrawable)
 {
+  UnsignedInteger drawableCount = drawablesCollection_.getSize();
   drawablesCollection_.add(aDrawable);
+  Drawable& drawable = drawablesCollection_[drawableCount];
```

Comment:
Is this "&" really necessary here?
```suggestion
  Drawable drawable = drawablesCollection_[drawableCount];
```

Reply:
Apparently changing it does matter: it breaks the unit test.

Reply:
My guess is that without &, you activate the copyAndWrite mechanism, so what you end up with is a copy of the Drawable in the Graph whose color_ attribute you change, but the Drawable within the Graph stays the same with the original color.

Reply:
Bottom line: we should restore the &